Tests/DependencyInjection/AbstractDoctrineExtensionTest.php

```diff
@@ -126,6 +126,7 @@ public function testDbalLoadSingleMasterSlaveConnection()
                 'dbname' => 'mysql_db',
                 'host' => 'localhost',
                 'unix_socket' => '/path/to/mysqld.sock',
+                'driverOptions' => [],
```

Comment:
Should the file on line 114 be modified to add options in both the master and the slave, to make things more interesting? Without your change, what happens? Are the master's options copied to the slave?

Reply:
There is actually no such option as "master" configuration key/option. What's defined one level above slaves is what's master. And yes that's how this patch works. It copies everything from master to slaves except the things in the list.

Reply:
Indeed, I read the issue backwards. What if there is already an option at the slave level? Will it be overwritten? Should the configuration at https://github.com/doctrine/DoctrineBundle/blob/6f3f4d9e27327cff20c584b0818b78c406325d72/Tests/DependencyInjection/Fixtures/config/yml/dbal_service_single_master_slave_connection.yml#L2-L15 be modified to test what happens when there are options in both levels? I'd expect some kind of merge to happen, but I'm not sure which level should get the priority. Maybe the slave one, because otherwise, there would be a BC-break?

Reply:
I don't think that's necessary. I didn't change merging logic. Whatever slave specifies is kept for that slave.